File: buildSrc/src/main/java/com/unifidokey/gradle/util/PropertyUtil.java
package com.unifidokey.gradle.util;

import org.gradle.api.Project;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.UncheckedIOException;
import java.util.Properties;

public class PropertyUtil {

    public static void load(Project project, Object path){
        // Load secret.properties if it exists
        File file = project.file(path);
        if (file.exists()) {
            Properties props = new Properties();
            try {
                props.load(new FileInputStream(file));
                props.forEach((key, value)->{
                    project.getExtensions().getExtraProperties().set(key.toString(), value);
                });
            } catch (IOException e) {
                throw new UncheckedIOException(e);
            }
        }
    }
}

File: buildSrc/src/main/java/com/unifidokey/gradle/util/PropertyUtil.kt
package com.unifidokey.gradle.util

import org.gradle.api.Project
import java.io.FileInputStream
import java.io.IOException
import java.io.UncheckedIOException
import java.util.Properties

object PropertyUtil {
    fun load(project: Project, path: Any) {
        // Load secret.properties if it exists
        val file = project.file(path)
        if (file.exists()) {
            val props = Properties()
            try {
                props.load(FileInputStream(file))
                props.forEach { key: Any, value: Any? ->
                    project.extensions.extraProperties[key.toString()] = value
                }
            } catch (e: IOException) {
                throw UncheckedIOException(e)
            }
        }
    }
}